python/sqlflow_playground/playground_server_design.md

```diff
@@ -0,0 +1,136 @@
+# SQLFlow Playground Server
+
+SQLFlow Playground Server exposes a REST API service that enables users to share
+the resources in one playground cluster. Users can take advantage of SQLFlow by
+installing a small plugin on her/his Jupyter Notebook.
+
+This service is used to extend the SQLFlow Playground's capability, especially
+when we need to manage the resource in the k8s cluster. It clearly is not for
+those who use the playground in single-user mode or those who just want to connect
+to the SQLFlow server in the playground.
+
+## The Architecture
+
+**SQLFlow Playground Server** is a side-by-side service of our playground cluster.
+Now, it is designed as an http server which receives user login, creates DB
+resource, and so on. This server uses `kubectl` to manipulate the resource in
+the playground(a k8s cluster). It's in someway the gateway of the playground.
+As described in the below diagram, the interaction of the three subjects could
+be: Clients ask the playground server for some resource. The server authorizes
+the client and create the resource on the playground. The client connects to
+the playground and does train/predict tasks using the created resource.
+
+```
+   ----------------run task--------------->
+   |                                      |
+Clients <--> Playground Server <--> Playground
```

Comment:
```
client -> playground server
client -> sqlflowserver
```
maybe should replace the word playground to sqlflowserver. Playground is a set of services deployed on the k8s cluster.